org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/PDEMavenBundlePluginConfigurator.java

```diff
@@ -0,0 +1,174 @@
+import org.eclipse.m2e.core.internal.markers.MavenProblemInfo;
+import org.eclipse.m2e.core.internal.markers.SourceLocation;
+import org.eclipse.m2e.core.internal.markers.SourceLocationHelper;
+import org.eclipse.m2e.core.lifecyclemapping.model.IPluginExecutionMetadata;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.core.project.configurator.AbstractBuildParticipant;
+import org.eclipse.m2e.core.project.configurator.AbstractProjectConfigurator;
+import org.eclipse.m2e.core.project.configurator.ILifecycleMappingConfiguration;
+import org.eclipse.m2e.core.project.configurator.MojoExecutionBuildParticipant;
+import org.eclipse.m2e.core.project.configurator.MojoExecutionKey;
+import org.eclipse.m2e.core.project.configurator.ProjectConfigurationRequest;
+import org.eclipse.m2e.jdt.IClasspathDescriptor;
+import org.eclipse.m2e.jdt.IJavaProjectConfigurator;
+
+/**
+ * This configurator performs the following tasks:
+ * <ul>
+ * <li>Enable the PDE nature for this project to make PDE aware of this
+ * project</li>
+ * <li>Set the location of the "bundle-root" where PDE looks for the
+ * manifest</li>
+ * </ul>
+ */
+@SuppressWarnings("restriction")
+public class PDEMavenBundlePluginConfigurator extends AbstractProjectConfigurator implements IJavaProjectConfigurator {
+
+	private static final String FELIX_PARAM_MANIFESTLOCATION = "manifestLocation";
+	private static final String FELIX_PARAM_SUPPORTINCREMENTALBUILD = "supportIncrementalBuild";
+	private static final String FELIX_MANIFEST_GOAL = "manifest";
+	private static final String BND_PARAM_MANIFESTLOCATION = "manifestPath";
+	private static final String[] BND_MANIFEST_GOALS = { "bnd-process", "bnd-process-tests", "jar", "test-jar" };
+
+	@Override
+	public void configure(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+		List<MojoExecution> executions = getMojoExecutions(request, monitor);
+		List<MavenProblemInfo> problems = new ArrayList<MavenProblemInfo>();
+		boolean hasManifestExecution = false;
+		for (MojoExecution execution : executions) {
+			Plugin plugin = execution.getPlugin();
+			if (isFelix(plugin)) {
+				MavenProject mavenProject = request.getMavenProject();
+				IMaven maven = MavenPlugin.getMaven();
+				if (isFelixBundleGoal(execution)) {
+					Boolean supportIncremental = maven.getMojoParameterValue(mavenProject, execution,
+							FELIX_PARAM_SUPPORTINCREMENTALBUILD, Boolean.class, monitor);
+					if (supportIncremental == null || !supportIncremental.booleanValue()) {
+						SourceLocation location = SourceLocationHelper.findLocation(execution.getPlugin(),
+								"configuration");
+						MavenProblemInfo problem = new MavenProblemInfo(
+								"Incremental updates are currently disabled, set supportIncrementalBuild=true to support automatic manifest updates for this project.",
+								IMarker.SEVERITY_WARNING, location);
+						problems.add(problem);
+					}
+					hasManifestExecution = true;
```

Comment:
The following code would give the user a warning to use the latest Maven-Bundle-Plugin (assuming the mentioned fixes will be integrated there). Maybe the message could be better.
```suggestion
					MavenVersion version = MavenVersion.parseMavenString(execution.getPlugin().getVersion());
					if (version != null && version.toReleaseVersion().compareTo(new MavenVersion("5.1.4")) < 0) {
						SourceLocation location = SourceLocationHelper.findLocation(execution.getPlugin(), "version");
						problems.add(new MavenProblemInfo(
								"The Maven-Bundle-Plugin only integrates well with Eclipse-M2E with version 5.4.0 or later",
								IMarker.SEVERITY_WARNING, location));
					}
					
					hasManifestExecution = true;
```